src/main/java/com/imsweb/layout/LayoutFactory.java

```diff
@@ -233,6 +256,9 @@ public static synchronized void registerLayout(Layout layout) {
         // make sure the layout is valid
         if (layout == null)
             throw new RuntimeException("Provided layout instance is null");
+        //Layout ID's must be unique
+        if (_INTERNAL_LAYOUTS.get(layout.getLayoutId()) != null || _LAYOUTS.get(layout.getLayoutId()) != null)
```

Comment:
We never add a null value to the caches, so why not calling "containsKey" instead of actually getting the layout? The behavior should be the same, and it would look a bit more logical than getting a value that you don't actually use (other than checking for null).